package/boot/grub2/grub2-efi/Makefile

```diff
@@ -0,0 +1,46 @@
+include $(TOPDIR)/rules.mk
+include $(INCLUDE_DIR)/kernel.mk
+
+PKG_NAME:=grub-efi
+
+include ../common.mk
+
+TAR_OPTIONS:= --transform 's/grub-${PKG_VERSION}/${PKG_NAME}-${PKG_VERSION}/' $(TAR_OPTIONS)
+
+PKG_BUILD_DEPENDS:=grub2-efi/host
+CONFIGURE_ARGS += --with-platform=efi
+HOST_CONFIGURE_ARGS += --with-platform=efi
+HOST_CONFIGURE_ARGS += --with-suffix=efi
+
+define Package/grub2-efi
+$(call Package/grub2/Default)
+DEPENDS:=@EFI_IMAGES +kmod-efi-vars
+TITLE += (with EFI support)
+endef
+
+$(eval $(call HostBuild))
+$(eval $(call BuildPackage,grub2-efi))
+
+include $(INCLUDE_DIR)/kernel.mk
+
+define KernelPackage/efi-vars
+  SUBMENU:=$(OTHER_MENU)
+  TITLE:=EFI VARS modules
+  DEPENDS:=@EFI_IMAGES
+  KCONFIG:= \
```

Comment:
I think, the best place to do this is in target/linux/x86/modules.mk . We don't really need to modify the config-default to only support EFI and we don't want to append EFI stuff in a non EFI system.